extension/src/inscription-view/open-page.ts

```diff
@@ -22,20 +21,49 @@ function isUrl(absolutePath: string) {
   return /^https?:\/\//i.test(absolutePath);
 }
 
-function openUrlInBrowser(absolutePath: string) {
-  vscode.env.openExternal(vscode.Uri.parse(absolutePath));
+function getValideFilePath(pathString: string) {
+  if (fs.existsSync(pathString)) {
+    return pathString;
+  }
+  const projectFolder = getProjectFolder();
+  if (projectFolder && fs.existsSync(path.join(projectFolder, pathString))) {
+    return path.join(projectFolder, pathString);
+  }
+  return null;
+}
+
+async function openUrlInIntegratedBrowser(absolutePath: string) {
+  await executeCommand('engine.ivyBrowserOpen', vscode.Uri.parse(absolutePath));
 }
 
-function isDocumentPath(absolutePath: string) {
-  const normalizedPath = path.normalize(absolutePath);
-  if (fs.existsSync(normalizedPath)) {
-    const stats = fs.statSync(normalizedPath);
-    return stats.isFile();
+function openInExplorer(absolutePath: string | null) {
+  if (absolutePath) {
+    vscode.commands.executeCommand('vscode.open', vscode.Uri.file(absolutePath));
+  } else {
+    vscode.window.showInformationMessage('The entered url is not valid.');
   }
+}
 
-  return false;
+function getProjectFolder() {
```

Comment:
there is a command `ivyProjects.getIvyProjects` that gives you the available ivy project paths. I think you can simply check if activeTab.input starts with one of them.